jolie/src/main/java/jolie/runtime/embedding/JolieServiceLoader.java

```diff
@@ -45,9 +49,26 @@ public JolieServiceLoader( Expression channelDest, Interpreter currInterpreter,
 
 		System.arraycopy( options, 0, newArgs, 2, options.length );
 		System.arraycopy( ss, 0, newArgs, 2 + options.length, ss.length );
+		CommandLineParser commandLineParser = new CommandLineParser( newArgs, currInterpreter.getClassLoader(), false );
 		interpreter = new Interpreter(
-			newArgs,
 			currInterpreter.getClassLoader(),
+			commandLineParser.getInterpreterParameters(),
+			currInterpreter.programDirectory() );
+	}
+
+	public JolieServiceLoader( String code, Expression channelDest, Interpreter currInterpreter )
+		throws IOException {
+		super( channelDest );
+		InterpreterParameters interpreterParameters = new InterpreterParameters(
+			currInterpreter.optionArgs(),
+			currInterpreter.getInterpreterParameters().includePaths(),
+			currInterpreter.getInterpreterParameters().libUrls(),
+			new File( "#native_code" ),
```

Comment:
You should generate something different for each embedded piece of code, or the debugging messages won't make much sense.